### PR TITLE
[ip] Correlations as normalized conditional entropy

### PR DESCRIPTION
**Pull request by minafarid**
[Do not merge yet]

* Implemented normalized conditional entropy as a measure of correlation.
* Modified some learning parameters in example

Current performance:
```
Precision = 1.00, Recall = 0.67, Repairing Recall = 0.79, 
F1 = 0.80, Repairing F1 = 0.88, 
Detected Errors = 435, Total Errors = 509, 
Correct Repairs = 342, Total Repairs = 342, Total Repairs (Grdth present) = 342
```

### CODE REVIEW

File: examples/holoclean_repair_example.py [hunk 1]
```diff
@@ -37,8 +38,8 @@
 # 4. Repair errors utilizing the defined features.
 hc.setup_domain()
 featurizers = [
-    InitAttrFeaturizer(),
-    InitSimFeaturizer(),
+    InitAttrFeaturizer(learnable=False),
```
Comment (richardwu) [inline examples/holoclean_repair_example.py]: Are we fixing these in this commit? Maybe you should do it in a separate commit.

File: domain/domain.py [hunk 2]
```diff
@@ -40,29 +42,68 @@ def setup(self):
...
+                if x == y:
+                    corr[x][y] = 1.0
+                    continue
+
+                # Compute the conditional entropy H(x|y) = H(x,y) - H(y).
+                # H(x,y) denotes H(x U y).
+                # If H(x|y) = 0, then y determines x, i.e., y -> x.
+                # Use the domain size of x as a log base for normalization.
+                y_vals = data_df[y]
+                x_y_entropy = drv.entropy_conditional(x_vals, y_vals, base=x_domain_size)
```
Comment (richardwu) [inline domain/domain.py:101]: Hmmm, I'm not too sure about introducing a new package dependency here just for conditional entropy, considering the project doesn't seem to be very active.

Implementing conditional entropy here with MLE shouldn't be too difficult considering that `x_vals` and `y_vals` are stored into a DataFrame anyways.

**Review by richardwu — CHANGES_REQUESTED** [review]

File: examples/holoclean_repair_example.py (hunk 1 above)
Comment (minafarid) [reply examples/holoclean_repair_example.py]: Agreed.. Reverted

File: domain/domain.py (hunk 2 above)
Comment (minafarid) [reply domain/domain.py:101]: This is a first iteration to prove that conditional entropy works well for our needs. The project was last updated 2 months ago, so it is not very old. 
I agree that a better thing to do is to implement our own conditional entropy, but this was faster to implement. We can revisit this in the next dev meeting.

**Review by richardwu — APPROVED** [review]
